infoschema/inspection_schema_test.go

```diff
@@ -0,0 +1,100 @@
+// Copyright 2019 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package infoschema_test
+
+import (
+	"strings"
+
+	. "github.com/pingcap/check"
+	"github.com/pingcap/failpoint"
+	"github.com/pingcap/tidb/domain"
+	"github.com/pingcap/tidb/kv"
+	"github.com/pingcap/tidb/session"
+	"github.com/pingcap/tidb/sessionctx/variable"
+	"github.com/pingcap/tidb/store/mockstore"
+	"github.com/pingcap/tidb/util/testkit"
+	"github.com/pingcap/tidb/util/testleak"
+)
+
+type inspectionSuite struct {
+	store kv.Storage
+	dom   *domain.Domain
+}
+
+var _ = SerialSuites(&inspectionSuite{})
+
+func (s *inspectionSuite) SetUpSuite(c *C) {
+	testleak.BeforeTest()
+
+	var err error
+	s.store, err = mockstore.NewMockTikvStore()
+	c.Assert(err, IsNil)
+	session.DisableStats4Test()
+	s.dom, err = session.BootstrapSession(s.store)
+	c.Assert(err, IsNil)
+}
+
+func (s *inspectionSuite) TearDownSuite(c *C) {
+	s.dom.Close()
+	s.store.Close()
+	testleak.AfterTest(c)()
+}
+
+func (s *inspectionSuite) TestInspectionTables(c *C) {
+	tk := testkit.NewTestKit(c, s.store)
+	instances := []string{
+		"pd,127.0.0.1:11080,127.0.0.1:10080,mock-version,mock-githash",
+		"tidb,127.0.0.1:11080,127.0.0.1:10080,mock-version,mock-githash",
+		"tikv,127.0.0.1:11080,127.0.0.1:10080,mock-version,mock-githash",
+	}
+	fpName := "github.com/pingcap/tidb/infoschema/mockClusterInfo"
+	fpExpr := `return("` + strings.Join(instances, ";") + `")`
+	c.Assert(failpoint.Enable(fpName, fpExpr), IsNil)
+	defer func() { c.Assert(failpoint.Disable(fpName), IsNil) }()
+
+	tk.MustQuery("select * from information_schema.cluster_info").Check(testkit.Rows(
+		"pd 127.0.0.1:11080 127.0.0.1:10080 mock-version mock-githash",
+		"tidb 127.0.0.1:11080 127.0.0.1:10080 mock-version mock-githash",
+		"tikv 127.0.0.1:11080 127.0.0.1:10080 mock-version mock-githash",
+	))
+
+	// enable inspection mode
+	inspectionTableCache := map[string]variable.TableSnapshot{}
```

Comment:
I didn't see the interface to enable it. So it's actually disabled and can't be enabled by user?

Reply:
The `InspectionExec` will be implemented in #14114